### DISCUSSION TITLE
`cue vet` with bytes type

### DISCUSSION THREAD

**Discussion by joelanford** [cue-lang/cue#1155] [Q&A]
I'm hoping to use `cue vet` to vet a JSON struct that contains a base64 encoded string, so I'd like my cue schema to contain a `bytes` field. However `cue vet` interprets my JSON struct as a `string` (technically, it is) and complains that it is not a `bytes` (even though it was exported from a `bytes` field). Is there any way to vet base64 encoded strings as bytes?

Given the follow files:

_schema.cue_
```cue
package hello
#Bytes: {
   base64data: bytes
}
```

_bytes.cue_
```cue
package hello
#Bytes & {
    base64data: 'Hello, World'
}
```

I expect to be able to export JSON and then vet it, but `vet` fails.
```console
$ cue export schema.cue bytes.cue > bytes.json
$ cat bytes.json
{
  "base64data": "SGVsbG8sIFdvcmxk"
}
$ cue vet bytes.json schema.cue -d '#Bytes'
base64data: conflicting values "SGVsbG8sIFdvcmxk" and bytes (mismatched types string and bytes):
    ./bytes.json:2:17
    ./schema.cue:2:14
```

**Comment by myitcv** [comment]
@joelanford welcome to the CUE project!

This looks like a bug, or more specifically a feature we haven't yet implemented totally in `cmd/cue`. This is very much analogous to recent work in the protobuf space, where data can be interpreted (imported/vetted/whatever) [in the context of a specific schema](https://pkg.go.dev/cuelang.org/go@v0.4.1-0.20210721131545-370fac9b4933/encoding/protobuf/textproto#Decoder.Parse). Here the transformation from JSON string field to bytes is something that could be expressed by the `-d` flag, and hence checked.

Please can you raise a feature request issue for this?

**Comment by amilicms** [comment]
Hello, I've also run into this issue with CUE `v0.8.0`. I would expect that the `bytes` would interpret the string as base64 and validate accordingly. My current work around is something along the lines of the following:
```cue
import "encoding/base64"

#base64Bytes: input={
	string
	#valid: base64.Decode(null, input)
}

b: #base64Bytes & "aGVsbG8sIHdvcmxkIQo="
// c: #base64Bytes & "$$$" // uncomment to see error
```

Ideally I would like to be able to use the `bytes` type directly.
